test/components/componentTests.ts

```diff
@@ -1,74 +1,87 @@
 ///<reference path="../testReference.ts" />
 
-function assertComponentXY(component: Plottable.Component, x: number, y: number, message: string) {
-  // use <any> to examine the private variables
-  let translate = d3.transform((<any> component)._element.attr("transform")).translate;
-  let xActual = translate[0];
-  let yActual = translate[1];
-  assert.strictEqual(xActual, x, "X: " + message);
-  assert.strictEqual(yActual, y, "Y: " + message);
-}
-
-describe("Component behavior", () => {
-  let svg: d3.Selection<void>;
+describe("Component", () => {
   let c: Plottable.Component;
   let SVG_WIDTH = 400;
   let SVG_HEIGHT = 300;
+  let svg: d3.Selection<void>;
   beforeEach(() => {
-    svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
     c = new Plottable.Component();
+    svg = TestMethods.generateSVG(SVG_WIDTH, SVG_HEIGHT);
   });
 
-  describe("anchor()", () => {
-    it("anchor()-ing works as expected", () => {
+  describe("anchoring", () => {
+    it("adds itself as a child element of the input selection", () => {
+      assert.strictEqual(c.anchor(svg), c, "setter returns calling object");
+      assert.strictEqual(Plottable.Utils.DOM.boundingSVG(<SVGElement> c.content().node()),
+        svg.node(), "component child of svg");
+      c.destroy();
+      svg.remove();
+    });
+
+    it("adds default selections in the correct order", () => {
+      c.anchor(svg);
+      assert.isFalse(c.foreground().empty(), "foreground exists in the DOM");
+      assert.isFalse(c.background().empty(), "background exists in the DOM");
+      assert.isFalse(c.content().empty(), "content exists in the DOM");
+      assert.isFalse(c.background().select(".background-fill").empty(), "background fill container exists in the DOM");
+
+      let componentElement = svg.select(".component");
+      let containers = componentElement.selectAll("g");
+      assert.strictEqual(containers[0][0], c.background().node(), "background at the back");
+      assert.strictEqual(containers[0][1], c.content().node(), "content at the middle");
+      assert.strictEqual(containers[0][2], c.foreground().node(), "foreground at the front");
+      assert.strictEqual(containers[0][3], componentElement.select(".box-container").node(), "boxes at front of foreground");
+      c.destroy();
+      svg.remove();
+    });
+
+    it("classes the input with 'plottable' if it is an svg", () => {
```

Comment:
It'd be nice to be more consistent.  In which case, I vote staying consistent with double quotes for the whole and single quotes for inside, since I personally find that more readable.  We should, however, get a poll in case people have better reasonings.